mungegithub/github/github.go

```diff
@@ -738,7 +709,31 @@ func (config *Config) SetBranchProtection(name string, contexts []string) error
 		return nil
 	}
 
-	return config.setBranchProtection(name, contexts)
+	request := &github.ProtectionRequest{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			Strict:   false,
+			Contexts: contexts,
+		},
+		RequiredPullRequestReviews: prot.RequiredPullRequestReviews,
+		Restrictions:               unchangedRestrictionRequest(prot.Restrictions),
+		EnforceAdmins:              false,
+	}
+	return config.setBranchProtection(name, request)
+}
+
+// unchangedRestrictionRequest generates a request that will
+// not make any changes to the teams and users that can merge
+// into a branch
+func unchangedRestrictionRequest(restrictions *github.BranchRestrictions) *github.BranchRestrictionsRequest {
+	request := &github.BranchRestrictionsRequest{}
+
+	for _, user := range restrictions.Users {
```

Comment:
`Teams and users restrictions are only available for organization-owned repositories.`

If this is nullable, wrap in an if-guard?